crates/handlers/src/graphql/model/site_config.rs

```diff
@@ -47,12 +51,33 @@ pub struct SiteConfig {
     /// Whether passwords are enabled and users can change their own passwords.
     password_change_allowed: bool,
 
+    /// Whether passwords are enabled and users can register using a password.
+    password_registration_enabled: bool,
+
     /// Minimum password complexity, from 0 to 4, in terms of a zxcvbn score.
     /// The exact scorer (including dictionaries and other data tables)
     /// in use is <https://crates.io/crates/zxcvbn>.
     minimum_password_complexity: u8,
 }
 
+#[derive(SimpleObject)]
+#[graphql(complex)]
+pub struct CaptchaConfig {
+    /// Which Captcha service is being used
+    pub service: CaptchaService,
+
+    /// The site key used by the instance
+    pub site_key: String,
+}
+
+/// Which Captcha service is being used
+#[derive(Enum, Debug, Clone, Copy, PartialEq, Eq)]
+pub enum CaptchaService {
+    RecaptchaV2,
+    CloudflareTurnstile,
+    HCaptcha,
+}
```

Comment:
I wondered whether this should be an enum with the site_key in each variant, so that we can extend later with provider-specific options and in the frontend we'd have to match on the `__typename` instead

but because this will never be actual public API, that won't be hard to change

Reply:
had the same line of thought